src/main/java/picard/fingerprint/HaplotypeProbabilitiesUsingLogLikelihoods.java

```diff
@@ -26,33 +26,46 @@
 
 import picard.util.MathUtil;
 
-import static java.lang.Math.log10;
-
 /**
  * Represents the probability of the underlying haplotype using log likelihoods as the basic datum for each of the SNPs. By convention the
  * alleles stored for each SNP are in phase.
  *
  * @author Tim Fennell
  * @author Yossi Farjoun
  */
-abstract class HaplotypeProbabilitiesUsingLogLikelihoods extends HaplotypeProbabilities {
+abstract class HaplotypeProbabilitiesUsingLogLikelihoods extends HaplotypeProbabilities implements Cloneable {
 
     // some derived classes might need to incorporate accumulated data before logLikelihood is usable.
     // use the getter to allow these classes to calculate the likelihood from the data.
-    private final double[] loglikelihoods = new double[Genotype.values().length];
+    // not final so that clone works
+    private double[] loglikelihoods = new double[Genotype.values().length];
+
     private boolean likelihoodsNeedUpdating = true;
 
     // stored in order to reduce computation we store these partial results.
     // they need to be recalculated if loglikelihoodNeedsUpdating
+
     private double[] likelihoods = new double[Genotype.values().length];
     private double[] posteriorProbabilities = new double[Genotype.values().length];
-    private double[] shiftedLogPosteriors = new double[Genotype.values().length];
 
+    //normalized (likeihood * prior / normalization_factor)
+    private double[] shiftedLogPosteriors = new double[Genotype.values().length];
 
     public HaplotypeProbabilitiesUsingLogLikelihoods(final HaplotypeBlock haplotypeBlock) {
         super(haplotypeBlock);
     }
 
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        final HaplotypeProbabilitiesUsingLogLikelihoods c = (HaplotypeProbabilitiesUsingLogLikelihoods)super.clone();
+        c.likelihoods = this.likelihoods.clone();
```

Comment:
Is there a reason why you don't use `System.arrayCopy()` here too? Or call `setLogLikelihoods()`? it would be nice to keep the field `final` if possible.

Reply:
super.clone does a shallow copy...in order to have these arrays be actually different, I need to clone them. I could make `new` arrays and arrayCopy, but I'm not sure why that is better. I don't think it's possible to keep the field final...not the clone, anyway...but I'm happy to learn something new.

Reply:
It's actually, the other way around...I need to get rid of the other arrayCopy and convert that to a clone..but I'll write a test the proves it first.